pkgs/build-support/trivial-builders/default.nix

```diff
@@ -410,7 +415,9 @@ rec {
         in
         if checkPhase == null then ''
           runHook preCheck
+        '' + lib.optionalString (runtimeShell == stdenv.shell) ''
           ${stdenv.shellDryRun} "$target"
+        '' + ''
```

Comment:
This seems a bit off to me. I guess this should be `runtimeShell == runtimeShell'`, but I also don't really know whether this even makes sense.

Since you didn't give a motivation for needing to customise `runtimeShell`, how about just not having this feature?

Reply:
I think this makes perfect sense. Otherwise we can just rename it to `writeBashApplication`.

Reply:
Actually you make a really good point, we _should_ rename it to `writeBashApplication`. This is really a bash-specific function, it doesn't support arbitrary shells. I think that's just a misnomer coming from `stdenv.runtimeShell`, which should also be renamed to `runtimeBash`, because it's never anything other than bash (and if you try, Nixpkgs will break in a million ways).

Instead, if different shells were to be supported, we can have separate functions for those, there's no universal "shell" syntax anyways that would make sense here.

Reply:
Yes, all that. But this PR *does* make it configurable.

Reply:
It sure does, but so far the only reason for having this is "because the function has `Shell` in the name", which is not good enough imo!

What _could_ make more sense is to explicitly call the function argument here `bash`, implying that people are expected to pass a valid bash. The implementation would then not have to rely on `stdenv.runtimeShell` nor `stdenv.shellDryRun` and instead use some splicing shenanigans and manually add the dry run arguments.